### ISSUE TITLE
Vision subsystem integration

### ISSUE THREAD

**Issue by kevlam2706** [FRC2706/2017-2706-Robot-Code#42]
We need to get the vision subsystem up and running once the practice robot is available.

- [x] ensure the end-to-end communication path works between the IP camera, Raspberry Pi, and RoboRio. (mostly done, see pull request #39)
- [x] add methods in the Camera subsystem to parse the received target data and turn that into navigation parameters (probably heading).
- [x] ensure that it is possible to ssh into the Pi to modify OpenCV configuration values in the pits. This may mean creating a separate ssh account on each of the TrackerBox2 Pi's.
- [ ] ensure that we have ssh client tools (like PuTTY) on our team laptops, especially the driver station.
- [x] controls team will find the best place to mount the camera and ring light
- [ ] create autonomous command groups to pull it all together!

**Comment by ounsworth** [comment]
Also worth science'ing if we need to come to a dead-stop in order to use the camera (like we did last year), or if the switch to a Centre-Of-Mass tracking algorithm means we're robust enough to motion blur that we can grab frames while moving.

(probably best to do this with a laptop acting as trackerbox so you can see the GUI output of TB)

**Comment by kevlam2706** [comment]
I pulled Apple Pi off the competition robot before it went into the bag. I tried connecting to it using the tools available to me at Enertron, which is more or less what we would have in a pit. I couldn't get in, so we'll need to reconfigure the Pi before we put it back on the robot.

**Comment by Lockscout** [comment]
How were you trying to connect to it?  Ssh, remote desktop or just a screen and keyboard? 

-Ethan

**Comment by kevlam2706** [comment]
ssh would be easiest. That way we could do it from the driver station laptop while sitting in the field looking at the targets under the correct lighting conditions. Otherwise we need to bring a monitor and keyboard with us.

**Comment by ounsworth** [comment]
Uhh, problem: to calibrate, you really really *really* need the trackerbox
GUI. Not something you can do easily from the command-line over ssh.


I would suggest this on-field calibration procedure:

- grab either my laptop, or a team laptop running ubuntu
- Start the trackerbox process
- set it to the same static IP as the Pi
- unplug the ethernet cable from the Pi and plug the laptop into the
robot's LAN
- calibrate trackerbox using the GUI
- ssh (or scp) the trackerbox calibration yaml file over to the Pi

**Comment by kevlam2706** [comment]
Fair point. Thanks Mike. So to follow the steps that Mike has outlined, we will use ssh to update the values on the Pi once we've determined what they should be.

 What I don't want is to be running keyboards and HDMI cables into the robot, or trying to pry the microSD card out of the Pi, or to have to unclip the Pi off the robot. We ended up doing some of these things last year and we can avoid the whole situation if we can ssh in to edit those settings.
